.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,29 @@
+name: build_commits
+
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: build commit code
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '^1.16'
+      - name: install llvm and libelf
+        run: |
+          sudo apt-get install llvm
+          sudo apt-get install libelf-dev
+      - name: git fetch commit code
+        run: |
+          git fetch origin pull/$PR/head:rebase-head
+          git checkout rebase-head
+        env:
+          PR: ${{ github.event.pull_request.number }}  
+      - name: build commits
+        run: git rebase --exec "make clean && make -j $(nproc)" rebase-head
+      - name: build all BPF datapath permutations
+        run: make --quiet -C bpf build_all
```

Comment:
This step is already performed by another GH action today: https://github.com/cilium/cilium/blob/d27cc1c66be3faa78822baf3530bb7d6dddc5b07/.github/workflows/bpf-checks.yaml#L56

I think it would be useful to run `git rebase --exec 'make -C bpf build_all'` though.

Reply:
Sure, I will make the required changes.